compose/config/validation.py

```diff
@@ -162,10 +163,25 @@ def _parse_oneof_validator(error):
         Inspecting the context value of a ValidationError gives us information about
         which sub schema failed and which kind of error it is.
         """
+
+        required = [context for context in error.context if context.validator == 'required']
+        if required:
+            return required[0].message
+
+        additionalProperties = [context for context in error.context if context.validator == 'additionalProperties']
+        if additionalProperties:
+            invalid_config_key = _parse_key_from_error_msg(additionalProperties[0])
+            return "contains unsupported option: '{}'".format(invalid_config_key)
+
         constraint = [context for context in error.context if len(context.path) > 0]
         if constraint:
             valid_types = _parse_valid_types_from_validator(constraint[0].validator_value)
-            msg = "contains {}, which is an invalid type, it should be {}".format(
+            path = [
```

Comment:
Aaaah, I see what's happening. I was focussed on `extends`, which being a dict would work happily, but the fields you've chosen in the tests highlights their `oneOf` contains a type of list with a format of items within that list. Right. Gotcha. Good catch :+1: 

Ok. I think we can make it a little clearer what's going on, I know the validator constraint is called `path`, what we're doing is trying to find the config key that contains something invalid. So how about something like:

```
            invalid_config_key = "".join(
                ["'{}' ".format(key) for key in constraint[0].path
                    if isinstance(key, six.string_types)
                ]
            )
```